### PR TITLE
Add async register method

### PR DESCRIPTION
**Pull request by keith-waters**
This PR does a few things:

- Moves registerServiceWorker to it own async method 
  - This is a breaking change
- Allows you to pass an `mapScriptUrl` to `register` 
  - then register passes it to `service-worker-loader`
- Move `node` dependency to dev dependencies 
- Bump to version 2, because of breaking changes
- Update the readme   

Thanks for a great repo! It's allowing us to download images and zip images with only the browser. :)

### CODE REVIEW

**Comment by robbederks** [comment]
Thanks for the PR. Always good to be as generic as possible, and I think the async register function will only help in the future.
Not very familiar with the `scriptUrl`, and in which scenarios you'd want to change it from the default value. Could you elaborate that a bit?
Will merge and push a new version to npm when when I find some time :)

**Comment by keith-waters** [comment]
With regards to `scriptUrl`: I was able to change the service worker url locally during testing. (Our frontend and backend code are served on different ports in dev env)

**Comment by keith-waters** [comment]
Realized `mapScriptUrl` will not work until `service-worker-loader` is updated. Triple checking to make sure it works before adding it here. 😬  Thanks for your patience!

File: package.json [hunk 1]
```diff
@@ -35,13 +35,13 @@
     "babel-plugin-istanbul": "^6.0.0",
     "mocha": "^6.2.3",
     "mocha-steps": "^1.3.0",
+    "node": "^13.14.0",
     "nyc": "^15.1.0",
-    "service-worker-loader": "^3.3.1",
+    "service-worker-loader": "^4.0.2",
```
Comment (keith-waters) [inline package.json:40]: There it is. Should be good now. :)

**Comment by robbederks** [comment]
Thx! Rebased and rebuilt this, and merged in (see PR #19). Will also push the new version to NPM

**Pull request closed without merging**